### PR TITLE
[expo-speech] Add support for max length of input string

### LINKED ISSUE #7214: [Expo-Speech] content limitation on Android only?
There seem to be a content limitation (in characters or size) on Android only. Can you please confirm? For example, that's how far I can go - add another word and it won't play:

```
    Speech.speak(
            "Asian markets mixed: Japan skids; China helped by rate cutMarkets are mixed in Asia, with Japan's benchmark slipping 0.7% after the government reported the economy contracted in the last quarter - - BANGKOK -- Markets were mixed in Asia on Monday, with Japan's benchmark slipping 0.7% after the government reported the economy contracted 6.3% in annual terms in the last quarter. China's shares got a boost after the central bank stepped in to help the economy with a rate cut, extra buying of securities and tax cuts. The Nikkei 225 in Tokyo closed at 23,523.24, while Sydney's S&P ASX/200 edged 1% lower to 7,125.10. South Korea's Kospi was flat, at 2,243.45, while the Hang Seng in Hong Kong climbed 0.6% to 27,990.46. India's Sensex edged 0.1% lower to 41,236.32. The Shanghai Composite index jumped 1.9% to 2,970.91 after the central bank and government announced a slew of measures to support the economy as the country battles an outbreak of a new virus that has killed 1,774 people  and infected more than 71,000 worldwide.Shares fell in Taiwan, rose in Bangkok and were flat in Jakarta and Singapore. The contraction in the Japanese economy, the world's third-largest, reflected the impact of typhoons, trade tensions and crimped consumer spending after the sales tax rose to 10% from 8% as of Oct. 1. The seasonally adjusted economic data was announced as Prime Minister Shinzo Abe faces pressure over spreading cases of the new viral illness COVID-19 and markets around the region see a mounting toll from its impact on travel and tourism as authorities strive to contain it. \"Consumer spending, which slumped following the tax hike in the fourth quarter of 2019, will now struggle to do anything except contract further in the first quarter as the impact of Covid-19 weighs on consumer sentiment, weighing in particular on the consumer services sector,\" ING said in a report. \"Some further government spending may help to curb any further contraction in GDP beyond 1Q20. But that will not stop what started off as a technical downturn from evolving into a full-blown recession,\" it said. Thailand and Singapore also reported weak growth in the last quarter of 2019, in figures that like Japan's do not yet reflect the consequences of the coronavirus outbreak. But good news came in the form of fresh help from the People's Bank of China, which cut its one-year medium-term lending rate to 3.15% from 3.25%. The central bank also injected some 200 billion yuan ($28.6 billion) and conducted 100 billion yuan ($14.3 billion) in reverse repos, in effect putting more cash into the market through short-term purchases of securities.  - - Such moves will likely be followed by still more, said Julian Evans-Pritchard, given that many of the companies worst affected by the virus outbreak are smaller ones that lack access to loans from major state-run banks. The government has also announced plans for tax cuts and other measures to help companies struggling with shut-downs of cities and plunging consumer spending and travel. \"We think the PBOC will need to expand its re-lending quotas and relax constraints on shadow banking in order to direct more credit to struggling SMEs,\" Evans-Pritchard said in a commentary. Wall Street closed out a wobbly day of trading Friday with the major stock indexes notching their second straight weekly gain. Though trading was mostly subdued and cautious following China's report Thursday of a surge in cases of a new virus that raised fresh concerns about global economic growth.The S&P 500 index rose 0.2% to 3,380.16. The Nasdaq composite gained 0.2%, to 9,731.18. Both indexes were lower for most of the afternoon. The Dow dropped 0.1%, to 29,398.08. The Russell 2000 index slid 0.4%, to 1,687.58.Technology companies led the gainers Friday. Chipmaker Nvidia was a standout, jumping 7% after it handily beat analysts profit forecasts for the fourth quarter.The real estate and utilities sectors also held up well as government bond yields fell, ",
            {
              voice: "com.apple.ttsbundle.Samantha-compact",
              language: "en",
              pitch: 1,
              rate: 1,
              onError: err => {},
            }
          );
```

### PR DESCRIPTION
**Pull request by barthap**
# Why
Fixes https://github.com/expo/expo/issues/7214

Also, maximum input string length is platform-dependent, so it would be useful to have a way to get it.

# How
Added new constant `Speech.maxSpeechInputLength`. It is platform dependent:
 - **Android** - used built-in [`getMaxSpeechInputLength()`](https://developer.android.com/reference/android/speech/tts/TextToSpeech#getMaxSpeechInputLength()) and exported it as a constant
- **iOS** - I haven't found anything about the limit, the constant returns `Number.MAX_VALUE` on TypeScript side
- **web** - [According to MDN](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisUtterance/text), it is 32767.

On Android and Web, calling `Speech.speak()` with text exceeding its maximum length, throws `ERR_SPEECH_INPUT_LENGTH`.

I also updated `expo-speech` docs with features above.

# Test Plan
It was tested on all platforms using code from [this snack](https://snack.expo.io/@barthap10/expo-speech-text-length-demo).
Test-suite ran without errors.

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/speech.md [hunk 1]
```diff
@@ -85,6 +102,12 @@ Speak out loud the `text` given `options`. Calling this when another text is bei
   - **onStopped (_function_)** -- A callback that is invoked when speaking is stopped by calling `Speech.stop()`.
   - **onError (_function_)** -- (Android only). A callback that is invoked when an error occurred while speaking.
 
+#### Error Codes
+
+- [`ERR_SPEECH_INPUT_LENGTH`](#err_speech_input_length)
+
+---
```
Comment (lukmccall) [inline docs/pages/versions/unversioned/sdk/speech.md:109]:
```suggestion
```

**Review by lukmccall — APPROVED** [review]
LGTM 🎉

File: docs/pages/versions/unversioned/sdk/speech.md (hunk 1 above)
Comment (barthap) [reply docs/pages/versions/unversioned/sdk/speech.md:109]: I think this separator increases readability...

Comment (lukmccall) [reply docs/pages/versions/unversioned/sdk/speech.md:109]: For me, it's looking weird, cause other methods aren't separate. It's looking more like the end of the "methods section".

**Pull request merged**